### PR TITLE
Support for self-hosted git repos

### PR DESCRIPTION
**Pull request by alexcb**
This allows users to use custom repo references in their `~/.earthly/config.yml`:
```
git:
  myserver:
    pattern: 'myserver/([^/]+)'
    substitute: 'ssh://root@$ip:2222/root/my/really/weird/path/\$1.git'
    auth: ssh
```

This would allos a user to reference a build like `earth myserver/project+docker`, which would end up cloning `ssh://root@$ip:2222/root/my/really/weird/path/project.git`

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -109,6 +109,30 @@ jobs:
...
+      FORCE_COLOR: 1
+      EARTHLY_TOKEN: "${{ secrets.EARTHLY_TOKEN }}"
+      SSH_PORT: "2222"
+    steps:
+      - uses: actions/checkout@v2
+      - name: install sshpass
+        run: sudo apt-get install -y sshpass
+      - name: Download released earth
+        run: "sudo /bin/sh -c 'wget https://github.com/earthly/earthly/releases/latest/download/earth-linux-amd64 -O /usr/local/bin/earth && chmod +x /usr/local/bin/earth'"
+      - name: Build latest earth using released earth
```
Comment (vladaionescu) [inline .github/workflows/ci.yml:131]: Don't forget docker login (see other tests).

File: .tests/self-hosted-private-repo.sh [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+EOF
+
+if ! $earth myserver/project:trunk+docker; then
+    docker ps -a
+    docker logs earthly-buildkitd
+    exit 1
+fi
+
+# finally test that the container was built and runs
+docker run --rm weirdrepo:latest | grep "hello weird world"
```
Comment (vladaionescu) [inline .tests/self-hosted-private-repo.sh:85]: Maybe place under `scripts/tests`?

Comment (vladaionescu) [inline .tests/self-hosted-private-repo.sh:85]: Add this file to shellcheck list?

**Review by vladaionescu — COMMENTED** [review]
Neat!

File: .tests/self-hosted-private-repo.sh (hunk 2 above)
Comment (alexcb) [reply .tests/self-hosted-private-repo.sh]: that's a good idea, `.tests` is a bit too hidden.

**Review by vladaionescu — APPROVED** [review]

**Pull request merged**